File: frontend/src/lib/api/inventory.ts
import { apiClient } from '../axios';

export interface InventoryProduct {
  id: string;
  sku: string;
  name: string;
  description?: string;
  category: string;
  unit_price: number;
  stock_quantity: number;
  min_stock_level: number;
  created_at: string;
  updated_at: string;
}

export const getInventory = async (): Promise<InventoryProduct[]> => {
  const response = await apiClient.get('/inventory/');
  const data = response.data?.data;
  if (data && Array.isArray(data.data)) {
    return data.data;
  }
  return Array.isArray(data) ? data : [];
};

export const createProduct = async (data: Partial<InventoryProduct>) => {
  const response = await apiClient.post('/inventory/', data);
  return response.data;
};

export const deleteProduct = async (id: string) => {
  const response = await apiClient.delete(`/inventory/${id}`);
  return response.data;
};

File: backend/models/inventory.py
from datetime import datetime
from typing import Optional
from beanie import Document
from pydantic import Field
from utils.helpers import utc_now

class InventoryProduct(Document):
    org_id: str
    sku: str = Field(..., description="Stock Keeping Unit / Item Code")
    name: str
    description: Optional[str] = None
    category: str
    unit_price: float = Field(default=0.0)
    stock_quantity: int = Field(default=0)
    min_stock_level: int = Field(default=5)
    
    created_at: datetime = Field(default_factory=utc_now)
    updated_at: datetime = Field(default_factory=utc_now)

    class Settings:
        name = "inventory_products"
